components/builder-worker/src/heartbeat.rs

```diff
@@ -12,224 +12,16 @@
-        HeartbeatMgr { state: PulseState::default(),
-                       pub_sock,
-                       cli_sock,
-                       heartbeat,
-                       msg: zmq::Message::new() }
-    }
-
-    // Main loop for server
-    fn run(&mut self,
-           rz: &mpsc::SyncSender<()>,
-           jobsrv_addrs: Vec<(String, String, String)>)
-           -> Result<()> {
-        for (hb, ..) in jobsrv_addrs {
-            println!("Connecting to heartbeat, {}", hb);
-            self.pub_sock.connect(&hb)?;
-        }
-        self.cli_sock.bind(INPROC_ADDR)?;
-        rz.send(()).unwrap();
-        // This hacky sleep is recommended and required by zmq for connections to establish
-        thread::sleep(Duration::from_millis(100));
-        let mut cli_sock_msg = false;
-        loop {
-            if self.state == PulseState::Pulse {
-                self.pulse()?;
-            }
-            {
-                let mut items = [self.cli_sock.as_poll_item(zmq::POLLIN)];
-                // Poll until timeout or message is received. Checking for the zmq::POLLIN flag on
-                // a poll item's revents will let you know if you have received a message or not
-                // on that socket.
-                zmq::poll(&mut items, HEARTBEAT_MS)?;
-                if items[0].is_readable() {
-                    cli_sock_msg = true;
-                }
-            }
-            if cli_sock_msg {
-                self.recv_cmd()?;
-                cli_sock_msg = false;
-            }
-        }
-    }
-
-    // Set internal state to `PulseState::Pause` and notify client OK
-    fn pause(&mut self) {
-        debug!("heartbeat paused");
-        self.state = PulseState::Pause;
-        self.cli_sock.send("", 0).unwrap();
-    }
-
-    // Broadcast to subscribers the HeartbeatMgr health and state
-    fn pulse(&mut self) -> Result<()> {
-        trace!("heartbeat pulsed: {:?}", self.heartbeat);
-        self.pub_sock.send(&message::encode(&self.heartbeat)?, 0)?;
-        Ok(())
-    }
-
-    // Wait receive for a command from a client
-    fn recv_cmd(&mut self) -> Result<()> {
-        self.cli_sock.recv(&mut self.msg, 0)?;
-        match self.msg.as_str() {
-            Some(CMD_PAUSE) => {
-                self.pause();
-                return Ok(());
-            }
-            Some(CMD_PULSE) => (),
-            _ => unreachable!("wk:hb:1, received unexpected message from client"),
-        }
-        self.cli_sock.recv(&mut self.msg, 0)?;
-        self.heartbeat = message::decode(&self.msg)?;
-        debug!("heartbeat received by mgr: {:?}", self.heartbeat);
-        self.resume();
-        Ok(())
-    }
-
-    // Set internal state to `PulseState::Pulse` and notify client OK
-    fn resume(&mut self) {
-        debug!("heartbeat resumed");
-        self.state = PulseState::Pulse;
-        self.cli_sock.send("", 0).unwrap();
-    }
-}
+pub fn worker_os() -> proto::Os { proto::Os::Darwin }
```

Comment:
if we are gutting the heartbeatcli, I don't think we need a separate file for what we are keeping.